### PR TITLE
Use the standard Bitcoin signed-message preimage in sign_message

### PR DESCRIPTION
**Pull request by SachinMeier**
## Summary

`Ecdsa.sign_message/2` hashed `"Bitcoin Signed Message:\n" <> msg` directly, omitting both CompactSize length prefixes required by the standard Bitcoin signed-message scheme. The correct preimage is:

```
compact_size(24) || "Bitcoin Signed Message:\n" || compact_size(byte_size(msg)) || msg
```

(the magic string is 24 bytes, so its prefix is the single byte `0x18`), and the digest is the double-SHA256 of that. The message-length prefix is also what makes the construction unambiguous.

For `msg = "hello"`:

| | digest |
|---|---|
| old (bitcoinex) | `5f03c0aeae12de40ebd6d15ef9c900aaefee6939fa8d051f231fe4e68674a568` |
| new (standard) | `cf0447ec85f0ce7150a257db32ebfcb7523dae17c36dbd1be598779fec0484f4` |

The old output was verifiable by **no other wallet** — signatures produced here could not be checked by Bitcoin Core, Electrum, etc., and theirs could not be checked here — defeating the function's only purpose (proof of key ownership).

## Changes

- `Ecdsa.message_digest/1` (new): returns the 32-byte digest of the standard preimage. Uses `Transaction.Utils.serialize_compact_size_unsigned_int/1` for the varints (`Utils.encode_int/1` was avoided deliberately: its `when i > 0` guard raises on the empty message).
- `Ecdsa.sign_message/2`: now signs `message_digest(msg)`.
- `Ecdsa.verify_message/3` (new): verifies a `(pubkey, msg, signature)` triple via public key recovery (`ecdsa_recover_compact/3`). Its absence is why this bug went unnoticed — nothing in the library ever verified a signed message.

## Tests

- Digest known-answer tests for `""` (the `encode_int` edge case), `"hello"`, and a 300-byte message (message-length varint in the `0xFD` two-byte form). Values derived from the spec by hand and cross-checked with Python `hashlib`.
- Regression guard: asserts the digest is not the old unprefixed one.
- Sign → recover round-trip across several messages, including empty and long.
- **Interop vector from Bitcoin Core** (`test/functional/rpc_signmessage.py`): WIF `cUeKHd5orzT3mz8P9pxyREHfsWtVfgsfDjiZZBcjUBAaGk1BTj7N` / address `mpLQjfK79b7CCV4VMJWEWAj5Mpx8Up5zxB` / message `"This is just a test message"` / signature `INbVnW4e6PeRmsv2Qgu8NuopvrVjkcxob+sX8OcZG0SALhWybUjzMLPdAsXI46YZGb0KQTRii+wWIQzRpG/U+S0=`. The fixed `sign_message/2` reproduces Core's signature byte for byte (RFC6979 + low-s), and pubkey recovery over `message_digest/1` returns Core's pubkey. The vector was independently validated offline (WIF checksum, address derivation, EC verification) before pinning.
- `verify_message/3` positives plus negatives (wrong message, wrong key, tampered signature).

All new tests fail against the unfixed code.

## Breaking change

**This is a breaking change** for anyone relying on the old (non-interoperable) digest: signatures produced by the previous `sign_message/2` will not verify against the new digest. Noted in `CHANGELOG.md` under Unreleased.

## Note on concurrent work

Two concurrent PRs touch neighbouring functions in the same module (`verify_signature/3` in `ecdsa.ex`, and `parse_signature/1`/`serialize_signature/1` in `secp256k1.ex`). This diff deliberately avoids modifying any of those — it only adds `message_digest/1` and `verify_message/3` and rewrites the body of `sign_message/2` — so the branches should merge cleanly.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: lib/secp256k1/ecdsa.ex [hunk 1]
```diff
@@ -101,22 +101,60 @@ defmodule Bitcoinex.Secp256k1.Ecdsa do
...
+  @doc """
+  verify_message verifies signature over message_digest/1 of msg by recovering
+  the public key and comparing it to pubkey.
+  """
+  @spec verify_message(Point.t(), binary, Signature.t()) :: boolean
+  def verify_message(%Point{} = pubkey, msg, %Signature{} = signature) do
+    digest = message_digest(msg)
+    # r and s must each occupy exactly 32 bytes for recovery to parse them
+    compact_sig =
+      Bitcoinex.Utils.int_to_big(signature.r, 32) <> Bitcoinex.Utils.int_to_big(signature.s, 32)
```
Comment (davidcaseria) [inline lib/secp256k1/ecdsa.ex]: `int_to_big/2` raises `ArgumentError` when `r` or `s` needs more than 32 bytes, because `Utils.pad/3` computes a negative pad length. `verify_message(pubkey, "hello", %Signature{r: 2 * n, s: 1})` raises instead of a `false` return, and the spec promises a boolean. Callers build `Signature` structs directly, as the test at line 370 of this PR does. Please reject an out-of-range `r` and `s` first, the same way `verify_signature/3` does in #125.

File: test/secp256k1/ecdsa_test.exs [hunk 2]
```diff
@@ -247,4 +247,164 @@ defmodule Bitcoinex.Secp256k1.EcdsaTest do
...
+      {:ok, privkey: privkey, pubkey: pubkey}
+    end
+
+    test "sign then recover pubkey from message digest", %{privkey: privkey, pubkey: pubkey} do
+      pubkey_hex = Point.serialize_public_key(pubkey)
+
+      for msg <- ["", "hello", "hello world", String.duplicate("a", 300)] do
+        sig = Ecdsa.sign_message(privkey, msg)
+        digest = Ecdsa.message_digest(msg)
+        compact_sig = Signature.serialize_signature(sig)
```
Comment (davidcaseria) [inline test/secp256k1/ecdsa_test.exs:296]: `serialize_signature/1` uses `:binary.encode_unsigned` without padding, so it returns fewer than 64 bytes when `r` or `s` has a leading zero byte. That happens for about 1 signature in 128. With this private key, the message "msg145" serializes to 63 bytes, and recovery then fails for all four recovery ids. This test passes only because its four messages produce full-length scalars.

Please build the compact signature with `Utils.int_to_big(sig.r, 32) <> Utils.int_to_big(sig.s, 32)`, to match `verify_message/3`. The root cause is `serialize_signature/1` itself, which also produces short Schnorr serializations. A fix there would let both this test and `verify_message/3` use it directly.

**Review by davidcaseria — CHANGES_REQUESTED** [review]
The preimage change is correct. I re-derived all three `message_digest/1` fixtures independently, and they match. Two issues inline, both about the 32-byte scalar padding that compact signatures need.
